nereid/nereid/api/api_v1/endpoints/network.py

```diff
@@ -36,161 +50,179 @@
             "nodes": [{"id": "A"}, {"id": "B"}],
             "edges": [{"source": "A", "target": "B"}],
         },
-        examples={}
-        # {
-        #     "valid":{
-        #         "graph": {
-        #             "directed": True,
-        #             "nodes": [{"id": "A"}, {"id": "B"}],
-        #             "edges": [{"source": "A", "target": "B"}],
-        #         }
-        #     },
-        #     "invalid":{
-        #         "graph": {
-        #             "directed": True,
-        #             "nodes": [{"id": "A"}, {"id": "B"}],
-        #             "edges": [{"source": "A", "target": "B"}],
-        #         }
-        #     },
-        # }
-        # ]
     )
 ):
+    start1 = time()
 
-    task = background_validate_network_from_dict.apply_async(
-        args=(jsonable_encoder(graph),)
-    )
+    task = background_validate_network_from_dict.apply_async(args=(graph.dict(),))
+
+    end1 = time()
+    logger.debug(f"elapsed time submit task: {end1-start1:.4f}")
 
+    start2 = time()
     try:
-        result = task.get(timeout=0.5)
+        _ = task.get(timeout=0.2)
     except celery.exceptions.TimeoutError:
         pass
+    end2 = time()
+    logger.debug(f"elapsed time try to get task: {end2-start2:.4f}")
+    logger.debug(f"cumul time: {end2-start1:.4f}")
 
+    response = dict(task_id=task.task_id, status=task.status)
+
+    start3 = time()
     if task.successful():
-        return dict(task_id=task.task_id, status=task.status, result=result)
+        response["data"] = task.result
 
     else:
-        result_path = router.url_path_for("get_network_result", task_id=task.id)
-
-        return dict(
-            task_id=task.id,
-            status=task.status,
-            result_route=f"{config.API_V1_STR}{result_path}",
+        result_path = router.url_path_for(
+            "get_validate_network_result", task_id=task.id
         )
+        response["result_route"] = f"{config.API_V1_STR}{result_path}"
+    end3 = time()
```

Comment:
i don't think so, but i could be convinced. since the task id is ephemeral and already logged by the `flower` service i dont think it's needed. this is really just meant to help the client find the result of a long-running task, so i'm not sure the server needs to write a log about it.

I propose we open a logging issue, and implement a coherent and useful logging pattern across each endpoint. it can include this idea, but i don't think this PR needs to wait for this to be solved. this route has some logging features that were helping me profile the function (wow, json serialization is 90% of the law), but a better logging pattern should be included in a future PR.

Reply:
OK, so the `else` block doesn't mean it failed. it means it's still processing